verilog/tools/kythe/verilog_kythe_extractor.cc

```diff
@@ -0,0 +1,73 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// verilog_kythe_extractor is a simple command-line utility to extract kythe
+// indexing facts from the given file.
+//
+// Example usage:
+// verilog_kythe_extractor files...
+
+#include "absl/flags/flag.h"
+#include "absl/status/status.h"
+#include "absl/strings/str_cat.h"
+#include "absl/strings/string_view.h"
+#include "absl/strings/substitute.h"
+#include "common/util/file_util.h"
+#include "common/util/init_command_line.h"
+#include "verilog/analysis/verilog_analyzer.h"
+#include "verilog/tools/kythe/indexing_facts_tree_extractor.h"
+
+ABSL_FLAG(bool, printextraction, false,
+          "Whether or not to print the extracted general indexing facts "
+          "from the middle layer)");
+
+int ExtractOneFile(absl::string_view content, absl::string_view filename) {
+  int exit_status = 0;
+  bool parse_ok = 0;
+
+  const auto& facts_tree =
+      verilog::kythe::ExtractOneFile(content, filename, exit_status, parse_ok);
+
+  // check for printextraction flag, and print extraction if on
+  if (absl::GetFlag(FLAGS_printextraction)) {
+    std::cout << std::endl
+              << (!parse_ok ? " (incomplete due to syntax errors): " : "")
+              << std::endl;
+
+    std::cout << facts_tree << '\n';
+  }
+
+  return exit_status;
+}
+
+int main(int argc, char** argv) {
+  const auto usage =
+      absl::StrCat("usage: ", argv[0], " [options] <file> [<file>...]");
```

Comment:
Add more of a description about what this program does, expected inputs/outputs, etc.

Reply:
Add at the top.